website/docs/plugin/framework/resources/state-move.mdx

```diff
@@ -0,0 +1,264 @@
+                    return
+                }
+
+                if req.SourceSchemaVersion != 0 {
+                    return
+                }
+
+                // This only checks the provider address namespace and type
+                // since practitioners may use differing hostnames for the same
+                // provider, such as a network mirror. If necessary though, the
+                // hostname can be used for disambiguation.
+                if !strings.HasSuffix(req.SourceProviderAddress, "examplecorp/examplecloud") {
+                    return
+                }
+
+                // Refer also to the RawState type JSON field which can be used
+                // with json.Unmarshal()
+                rawStateValue, err := req.SourceRawState.Unmarshal(tftypes.Object{
+                    AttributeTypes: map[string]tftypes.Type{
+                        "id":               tftypes.String,
+                        "source_attribute": tftypes.Bool,
+                    },
+                })
+
+                if err != nil {
+                    resp.Diagnostics.AddError(
+                        "Unable to Unmarshal Source State",
+                        err.Error(),
+                    )
+
+                    return
+                }
+
+                var rawState map[string]tftypes.Value
+
+                if err := rawStateValue.As(&rawState); err != nil {
+                    resp.Diagnostics.AddError(
+                        "Unable to Convert Source State",
+                        err.Error(),
+                    )
+
+                    return
+                }
+
+                var id *string
+
+                if err := rawState["id"].As(&id); err != nil {
+                    resp.Diagnostics.AddAttributeError(
+                        path.Root("id"),
+                        "Unable to Convert Source State",
+                        err.Error(),
+                    )
+
+                    return
+                }
+
+                var sourceAttribute *bool
+
+                if err := rawState["source_attribute"].As(&sourceAttribute); err != nil {
+                    resp.Diagnostics.AddAttributeError(
+                        path.Root("source_attribute"),
+                        "Unable to Convert Source State",
+                        err.Error(),
+                    )
+
+                    return
+                }
+
+                targetStateData := TargetResourceModel{
+                    Id:              types.StringPointerValue(id),
+                    TargetAttribute: types.BoolPointerValue(sourceAttribute),
+                }
+
+                resp.Diagnostics.Append(resp.TargetState.Set(ctx, targetStateData)...)
+            },
+        },
+    }
+}
+```
+
+## Caveats
+
+Note these caveats when implementing the `MoveState` method:
+
+* An error is returned if the response state contains unknown values. Set all attributes to either null or known values in the response.
```

Comment:
Curious q: Is this validation implemented on the Terraform core side?

Reply:
Looks like this should be good! https://github.com/hashicorp/terraform/blob/48148eda5f5111fe4d8fdd28882472076227691a/internal/refactoring/cross_provider_move.go#L165-L173